159 media gallery

paolomoz opened the pull request:
Fix #159

Original:
https://www.moleculardevices.com/products/microplate-readers/multi-mode-readers/spectramax-i3x-readers

Test URLs:
- Before: https://main--moleculardevices--hlxsites.hlx.page/drafts/paolom/spectramax-i3x-readers1
- After: https://159-media-gallery--moleculardevices--hlxsites.hlx.page/drafts/paolom/spectramax-i3x-readers1

andreituicu commented on blocks/hero-advanced/hero-advanced.js:
```diff
@@ -19,4 +51,15 @@ export default async function decorate(block) {
+    mg.parentElement.appendChild(overlay);
+    mg.addEventListener('click', () => {
+      openMediaGallery(overlay);
+    });
```
There are a few options for building the media gallery without doing it on click:
1. using an `async` method without an `await`. We have this in the hero block for the breadcrumbs: https://github.com/hlxsites/moleculardevices/blob/main/blocks/hero/hero.js#L105. by calling an async method, without an await, you are simply creating a promise.
2. `setTimeout(() => { createMediaGallery(overlay) }` without a specified delayed time. you can use the function for creating the media gallery in setTimeout which will be sending the creation to the first time the browser has capacity to process it.
3. `setTimeout(1000, () => { createMediaGallery(overlay))` with specified timeout. this will delays the creation of the media gallery by the specified time (e.g. 1s - 2s).

You can experiment with these. The only thing to look for is the LCP. none of the options above should be delaying the LCP, but just to be sure. :)

andreituicu replied:
just to mention that the way it is implemented at the moment on click, new elements are being created for each click, so if you click a 3 times `View Media Gallery`, it will create 3x the number of overlay contents

paolomoz replied:
done calling async method; so to open the media gallery we just add the class now. which also fixes the multiple click issue. thanks!

andreituicu commented on blocks/hero-advanced/hero-advanced.js:
```diff
@@ -19,4 +51,15 @@ export default async function decorate(block) {
   buildHero(block);
+
+  const mgs = block.querySelectorAll('a[href*="/media-gallery"]');
+  [...mgs].forEach((mg) => {
```
Can there be more than one media gallery?

paolomoz replied:
no; good point as we always tend to over engineer :) I changed it to support 1

paolomoz commented:
> There seems to be a carousel page counter beside the video description, e.g. `1 of 2 Novel Ways to Monitor Cell Growth and Cytotoxicity: StainFree Technology` in https://www.moleculardevices.com/products/microplate-readers/multi-mode-readers/spectramax-i3x-readers. We added carousel's page counter support in #260 so you can leverage the same code.

added counter (I was hoping to skip, but since you noted, I added it :) 
I didn't reuse the code as it took just 2 lines for me

> * There's a horizontal scroller at the bottom of the carousel which should be hidden

I cannot repro. Can you provide more details how to repro?

> * The page counter / description on top should be aligned left

moved, now it's similar to original (I thought it would be better at the center
